contracts/sdk/PublicSignalsBuilder.sol

```diff
@@ -0,0 +1,339 @@
+     * public signals array.
+     * @param dataPointer_ Pointer to the public signals array in memory.
+     * @param expirationDateLowerbound_ The lower bound expiration date value (encoded). Must be less
+     * than the passport expiration date (encoded `yyMMdd`).
+     * @param expirationDateUpperbound_ The upper bound expiration date value (encoded). Must be greater
+     * than the passport expiration date (encoded `yyMMdd`).
+     */
+    function withExpirationDateLowerboundAndUpperbound(
+        uint256 dataPointer_,
+        uint256 expirationDateLowerbound_,
+        uint256 expirationDateUpperbound_
+    ) internal pure {
+        assembly {
+            // 32 + 20 * 32 = 32 + 640 = 672
+            mstore(add(dataPointer_, 672), expirationDateLowerbound_)
+            // 32 + 21 * 32 = 32 + 672 = 704
+            mstore(add(dataPointer_, 704), expirationDateUpperbound_)
+        }
+    }
+
+    /**
+     * @notice Sets the citizenshipMask (index 22) in the public signals array.
+     * @param dataPointer_ Pointer to the public signals array in memory.
+     * @param citizenshipMask_ The citizenship mask value.
+     */
+    function withCitizenshipMask(uint256 dataPointer_, uint256 citizenshipMask_) internal pure {
+        assembly {
+            // 32 + 22 * 32 = 32 + 704 = 736
+            mstore(add(dataPointer_, 736), citizenshipMask_)
+        }
+    }
+
+    /**
+     * @notice Converts the public signals array to a uint256 array.
+     */
+    function buildAsUintArray(
+        uint256 dataPointer_
+    ) internal pure returns (uint256[] memory pubSignals_) {
+        assembly {
+            pubSignals_ := dataPointer_
+        }
+    }
+
+    /**
+     * @notice Converts the public signals array to a bytes32 array.
+     */
+    function buildAsBytesArray(
+        uint256 dataPointer_
+    ) internal pure returns (bytes32[] memory pubSignals_) {
+        assembly {
+            pubSignals_ := dataPointer_
+        }
+    }
+
+    /**
+     * @notice Validates the date by checking if it is within a 1-day range of the current block timestamp.
+     * @param parsedTimestamp_ The parsed timestamp value.
+     * @return true if the date is valid, false otherwise.
+     */
+    function validateDate(uint256 parsedTimestamp_) internal view returns (bool) {
+        // +- 1 day validity
+        return
+            parsedTimestamp_ > block.timestamp - 1 days &&
+            parsedTimestamp_ < block.timestamp + 1 days;
+    }
+
+    /**
+     * @notice Retrieves the AQueryProofVerifierBuilder.ABuilderStorage storage reference.
+     * @return $ The AQueryProofVerifierBuilder.ABuilderStorage storage reference.
+     */
+    function getABuilderStorage()
+        internal
+        pure
+        returns (AQueryProofVerifierBuilder.ABuilderStorage storage $)
+    {
+        assembly {
+            $.slot := A_BUILDER_STORAGE
+        }
+    }
+}
```

Comment:
Probably `private` functions.

Reply:
Not sure about getABuilderStorage, because it is needed in AQueryProofExecutor